### PR TITLE
test: Write tests for Checkbox component

### LINKED ISSUE #77: `Checkbox` component tests
## Description

Write Cypress tests for the `Checkbox` component

## Acceptance Criteria

* [ ] Writes tests for all usage scenarios
* [ ] Refactors component to use `copyContent`
* [ ] Updates `pull_request` workflow for new tests

### PR DESCRIPTION
**Pull request by Tristan-Holtz**
## Description

Tests written for `Checkbox` component

- Issues:
  - Resolves #77 

## Changes

- Write `Checkbox` component tests
- Refactor `Checkbox` stories with copy content

## Sanity Checks

- [x] There are no incidental style changes

### CODE REVIEW

**Comment by HartiganHM** [comment]
@Tristan-Holtz Updated the PR description to name the component explicitly. It's a little redundant to have "write tests" in the description and changes, but it's what I've been doing, so I updated it to be consistent.

It looks like all of the `Checkbox` tests are passing though! Is this ready to be merged in?

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -160,8 +167,8 @@ jobs:
       # Set an environment variable for the publishing dev
       # This is used in to make a unique canary release to npm
       PR_DEV: ${{ github.actor }}
-      UUID: ""
-      VERSION: ""
+      UUID: ''
+      VERSION: ''
```
Comment (HartiganHM) [inline .github/workflows/pull_request.yml:171]: We might have conflicting Prettier configs here. I'll have to double check.

File: src/cypress/integration/Form/Checkbox/checkbox-disabled.spec.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { checkboxCopy } from 'shared/data/copyContent';
+
+describe('checkbox-disabled-test', () => {
```
Comment (HartiganHM) [inline src/cypress/integration/Form/Checkbox/checkbox-disabled.spec.ts:3]: Let's update this to the way you have it in the `checkbox-default` tests. It should read: "Checkbox Disabled tests".

File: src/cypress/integration/Form/Checkbox/checkbox-disabled.spec.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      .should('be.disabled');
+  });
+
+  it('Should be not be checked', () => {
+    cy.findByText(checkboxCopy.disabled)
+      .siblings('input')
+      .should('not.be.checked');
+  });
+
+  it('Should be checked', () => {
```
Comment (HartiganHM) [inline src/cypress/integration/Form/Checkbox/checkbox-disabled.spec.ts:25]: Let's update these to be more specific. Something like:
* Should have both options disabled
* Should not have the first option checked
* Should have the second option checked

File: src/cypress/integration/Form/Checkbox/checkbox-error.spec.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { checkboxCopy } from 'shared/data/copyContent';
+
+describe('checkbox-error-tests', () => {
```
Comment (HartiganHM) [inline src/cypress/integration/Form/Checkbox/checkbox-error.spec.ts:3]: Let's update this describe block too.

File: src/cypress/integration/Form/Checkbox/checkbox-with-other-options.spec.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { checkboxCopy } from 'shared/data/copyContent';
+
+describe('checkbox-with-other-options-tests', () => {
```
Comment (HartiganHM) [inline src/cypress/integration/Form/Checkbox/checkbox-with-other-options.spec.ts:3]: Same here.

File: src/cypress/integration/Form/Checkbox/checkbox-with-other-options.spec.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    cy.get('.fd-other-option__input').should('have.value', expected);
+  });
+
+  it('Should remove value from "Other" if unchecked', () => {
+    cy.get('.fd-other-option')
+      .siblings('input')
+      .uncheck({ force: true });
+
+    cy.get('.fd-other-option__input').should('not.have.value');
+  });
```
Comment (HartiganHM) [inline src/cypress/integration/Form/Checkbox/checkbox-with-other-options.spec.ts:30]: Can you just leave a comment in here to update these once the `Checkbox` and `OtherOption` components are refactored? Something like:

```javascript
// TODO: Update to use findByLabel when Checkbox/OtherOption are refactored
```

This just helps to remind us to update it at some point, outside of the issue we made.

File: src/shared/data/copyContent/checkboxCopy.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  hedgehog: 'Hedgehog',
+  ocelot: 'Ocelot',
+  cthulhu: 'Cthulhu',
+  errorMessage: 'Please pick a favorite pet',
+  bestIceCream: 'Best Ice Cream',
+  other: 'Other',
+  chocolate: 'Chocolate',
+  vanilla: 'Vanilla',
+  saltedOreo: 'Salted Oreo',
+  cookieDough: 'Cookie Dough',
```
Comment (HartiganHM) [inline src/shared/data/copyContent/checkboxCopy.ts:22]: Alphabetize please 🙃

**Review by HartiganHM — CHANGES_REQUESTED** [review]
Overall, all the tests look great and they're passing in the pipeline 🎉 

There are a few small language/organization updates, but once those are in, we should be set!

File: src/cypress/integration/Form/Checkbox/checkbox-error.spec.ts (hunk 4 above)
Comment (Tristan-Holtz) [reply src/cypress/integration/Form/Checkbox/checkbox-error.spec.ts]: Got it

**Review by HartiganHM — DISMISSED** [review]
I went ahead and added those `TODO` notes to the "other option" tests and reverted the single-quotes to double-quotes in the `pull_request.yml`. That must be a Prettier misconfiguration on your machine, so we'll dive into that next.

All of the tests look great though! Excited to get these running in the pipeline and move on to the next one 🎉

**Review by HartiganHM — APPROVED** [review]

**Pull request merged**